### DISCUSSION TITLE
Best practice for getting identifiers for a ManyToMany property?

### DISCUSSION THREAD

**Discussion by SpencerKaiser** [mikro-orm/mikro-orm#3086] [Q&A] [answered]
I need to retrieve an entity via my API and the `ManyToMany` property for my entity isn't being populated as I expect; I want `Author.books` to be an array of IDs (as you'd get with a normal `ManyToOne`) but the property is just missing. I added it but then the response included the full objects... is there a way to _**just**_ get the ids?

`Author.ts`
``` typescript
// Author entity

@ManyToMany(() => Books, 'authors', { owner: true })
books = new Collection<Book>(this);
```

`authors/get.ts`
```typescript
const author = await em.findOne(
  Author,
  { id: 12 },
  { populate: ['books'] }, // Without this, `author` does not have the `books` property
);

author.books.populated(false); // Without this line, `books` is an array of entities, not identifiers
res.send(author);
```

**Comment by B4nan** [comment, marked as answer]
You obviously need to populate it, otherwise there is nothing - the information lives in pivot table and you cant expect we somehow get the list of IDs without you saying you want to populate it. Marking it as "not populated" afterwards has exactly this effect, so it gets serialized as an array of PKs only.

There is no way to load just the IDs, issue here https://github.com/mikro-orm/mikro-orm/issues/1158

**Reply by SpencerKaiser** [reply]
Awesome, thanks! I subscribed to that issue as well 👌
